prime-router/src/main/kotlin/Metadata.kt

```diff
@@ -51,28 +58,48 @@ class Metadata {
     private val mapper = ObjectMapper(YAMLFactory()).registerModule(KotlinModule())
 
     /**
-     * Load all parts of the metadata catalog from a directory and its sub-directories
+     * The database lookup table access.
      */
-    constructor(
-        metadataPath: String
-    ) {
+    private var tableDbAccess: DatabaseLookupTableAccess = DatabaseLookupTableAccess()
+
+    /**
+     * Load all parts of the metadata catalog located at the default catalog folder.
+     * @param tableDbAccess database lookup table access for dependency injection purposes
+     */
+    internal constructor(tableDbAccess: DatabaseLookupTableAccess? = null) :
+        this(defaultMetadataDirectory, tableDbAccess)
+
+    /**
+     * Load all parts of the metadata catalog located at [metadataPath] from a directory and its sub-directories.
+     * @param tableDbAccess database lookup table access for dependency injection purposes
+     */
+    internal constructor(
+        metadataPath: String,
+        tableDbAccess: DatabaseLookupTableAccess? = null
+    ) : this() {
+        if (tableDbAccess != null) this.tableDbAccess = tableDbAccess
         val metadataDir = File(metadataPath)
         if (!metadataDir.isDirectory) error("Expected metadata directory")
         loadValueSetCatalog(metadataDir.toPath().resolve(valuesetsSubdirectory).toString())
         loadLookupTables(metadataDir.toPath().resolve(tableSubdirectory).toString())
+        loadDatabaseLookupTables()
         loadSchemaCatalog(metadataDir.toPath().resolve(schemasSubdirectory).toString())
         loadFileNameTemplates(metadataDir.toPath().resolve(fileNameTemplatesSubdirectory).toString())
+        logger.trace("Metadata initialized.")
     }
 
     /**
      * Useful for test versions of the metadata catalog
+     * @param tableDbAccess database lookup table access for dependency injection purposes
```

Comment:
Would you mind adding an explicit comment that this is for testing only, and should never be called from any non-test code?

Reply:
Good idea.  Done.